src/realm/sync/subscriptions.cpp

```diff
@@ -829,15 +828,9 @@ std::vector<SubscriptionSet> SubscriptionStore::get_pending_subscriptions() cons
     std::vector<SubscriptionSet> subscriptions_to_recover;
     auto active_sub = get_active();
     auto cur_query_version = active_sub.version();
-    DB::version_type db_version = 0;
-    if (active_sub.state() == SubscriptionSet::State::Complete) {
-        db_version = active_sub.snapshot_version();
-    }
-    REALM_ASSERT_EX(db_version != DB::version_type(-1), active_sub.state());
     // get a copy of the pending subscription sets since the active version
-    while (auto next_pending = get_next_pending_version(cur_query_version, db_version)) {
```

Comment:
Is there ever a case we want to use `db_version`?
I can't think of any, since the goal is to progress through the subscriptions one by one and the subscription state should be up to date with whether or not it has been integrated.